packages/vega-parser/src/parsers/axis.js

```diff
@@ -39,12 +39,17 @@ export default function(spec, scope) {
   // encoding properties for axis group item
   axisEncode = extendEncode({
     update: {
-      offset:       encoder(_('offset') || 0),
+      offset:       encoder(_('offset', 0)),
```

Comment:
I believe this is a cleaner way to add defaults but I'm happy to revert if it's not correct.

Reply:
The prior version is safer against invalid inputs like empty string or false, but I'm not sure how critical that is.

Reply:
In the end, the new code uses this to use the default so yes, empty strings or false would be passed to offset but those are invalid values so 🤷‍♂. I guess there is no point in changing this so I'll revert. 

```
export function value(specValue, defaultValue) {
  return specValue != null ? specValue : defaultValue;
}
```

Reply:
The legend encode code doesn't seem to handle default values at all.

```js
addEncoders(encode, {
    orient:       _('orient'),
    offset:       _('offset'),
    padding:      _('padding'),
    titlePadding: _('titlePadding'),
    cornerRadius: _('cornerRadius'),
    fill:         _('fillColor'),
    stroke:       _('strokeColor'),
    strokeWidth:  config.strokeWidth,
    strokeDash:   config.strokeDash,
    x:            _('legendX'),
    y:            _('legendY'),
    ariaHidden:   ariaHidden,
    ariaLabel:    ariaHidden == true ? undefined : _('ariaLabel', legendAriaLabel(_, scope)),
    ariaRole:     ariaHidden == true ? undefined : _('ariaRole'),
    ariaRoleDescription: ariaHidden == true ? undefined : _('ariaRoleDescription'),
    tabindex:     _('tabindex')
  });
```